File: 367.ValidPerfectSquare/square.py
class Solution(object):
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num < 0:
            return False
        if num <= 1:
            return True
        root = 2
        while root <= num / root:
            if root * root == num:
                return True
            root += 1
        return False

if __name__ == '__main__':
    for i in range(101):
        if Solution().isPerfectSquare(i):
            print i

File: 367.ValidPerfectSquare/Square.java
class Solution {
    public boolean isPerfectSquare(int num) {
        if (num < 0) {
            return false;
        }
        if (num < 2) {
            return true;
        }
        int root = 2;
        while (root <= num / root) {
            if (root * root == num) {
                return true;
            }
            root++;
        }
        return false;
    }
}
